arch/PowerPC/PPCDisassembler.c

```diff
@@ -1,627 +1,515 @@
-	MCOperand_CreateImm0(Inst, SignExtend64(Imm, N));
-
+	if (RegNo & 1)
+		return MCDisassembler_Fail;
+	MCOperand_CreateReg0(Inst, (VSRpRegs[RegNo >> 1]));
 	return MCDisassembler_Success;
 }
 
-
-#define GET_INSTRINFO_ENUM
-#include "PPCGenInstrInfo.inc"
-
 static DecodeStatus decodeMemRIOperands(MCInst *Inst, uint64_t Imm,
-		int64_t Address, const void *Decoder)
+										int64_t Address, const void *Decoder)
 {
 	// Decode the memri field (imm, reg), which has the low 16-bits as the
 	// displacement and the next 5 bits as the register #.
 
 	uint64_t Base = Imm >> 16;
 	uint64_t Disp = Imm & 0xFFFF;
 
-	// assert(Base < 32 && "Invalid base register");
-	if (Base >= 32)
-		return MCDisassembler_Fail;
-
 	switch (MCInst_getOpcode(Inst)) {
-		default: break;
-		case PPC_LBZU:
-		case PPC_LHAU:
-		case PPC_LHZU:
-		case PPC_LWZU:
-		case PPC_LFSU:
-		case PPC_LFDU:
-				 // Add the tied output operand.
-				 MCOperand_CreateReg0(Inst, GP0Regs[Base]);
-				 break;
-		case PPC_STBU:
-		case PPC_STHU:
-		case PPC_STWU:
-		case PPC_STFSU:
-		case PPC_STFDU:
-				 MCInst_insert0(Inst, 0, MCOperand_CreateReg1(Inst, GP0Regs[Base]));
-				 break;
+	default:
+		break;
+	case PPC_LBZU:
+	case PPC_LHAU:
+	case PPC_LHZU:
+	case PPC_LWZU:
+	case PPC_LFSU:
+	case PPC_LFDU:
+		// Add the tied output operand.
+		MCOperand_CreateReg0(Inst, (RRegsNoR0[Base]));
+		break;
+	case PPC_STBU:
+	case PPC_STHU:
+	case PPC_STWU:
+	case PPC_STFSU:
+	case PPC_STFDU:
+		MCInst_insert0(Inst, 0, MCOperand_CreateReg1(Inst, RRegsNoR0[Base]));
+		break;
 	}
 
-	MCOperand_CreateImm0(Inst, SignExtend64(Disp, 16));
-	MCOperand_CreateReg0(Inst, GP0Regs[Base]);
-
+	MCOperand_CreateImm0(Inst, (SignExtend64(Disp, 16)));
+	MCOperand_CreateReg0(Inst, (RRegsNoR0[Base]));
 	return MCDisassembler_Success;
 }
 
 static DecodeStatus decodeMemRIXOperands(MCInst *Inst, uint64_t Imm,
-		int64_t Address, const void *Decoder)
+										 int64_t Address, const void *Decoder)
 {
 	// Decode the memrix field (imm, reg), which has the low 14-bits as the
 	// displacement and the next 5 bits as the register #.
 
 	uint64_t Base = Imm >> 14;
 	uint64_t Disp = Imm & 0x3FFF;
 
-	// assert(Base < 32 && "Invalid base register");
-	if (Base >= 32)
-		return MCDisassembler_Fail;
-
 	if (MCInst_getOpcode(Inst) == PPC_LDU)
 		// Add the tied output operand.
-		MCOperand_CreateReg0(Inst, GP0Regs[Base]);
+		MCOperand_CreateReg0(Inst, (RRegsNoR0[Base]));
 	else if (MCInst_getOpcode(Inst) == PPC_STDU)
-		MCInst_insert0(Inst, 0, MCOperand_CreateReg1(Inst, GP0Regs[Base]));
+		MCInst_insert0(Inst, 0, MCOperand_CreateReg1(Inst, RRegsNoR0[Base]));
 
-	MCOperand_CreateImm0(Inst, SignExtend64(Disp << 2, 16));
-	MCOperand_CreateReg0(Inst, GP0Regs[Base]);
+	MCOperand_CreateImm0(Inst, (SignExtend64(Disp << 2, 16)));
+	MCOperand_CreateReg0(Inst, (RRegsNoR0[Base]));
+	return MCDisassembler_Success;
+}
 
+static DecodeStatus decodeMemRIHashOperands(MCInst *Inst, uint64_t Imm,
+											int64_t Address,
```

Comment:
Same problem of huge empty spaces?

Reply:
@XVilka This depends on the tab width in Github or your editor. Github defaults to 8 spaces per tab.